From: Xin Hao <xhao@linux.alibaba.com>
Subject: mm/damon/core: remove duplicate check about THP
Date: Thu, 15 Sep 2022 22:22:37 +0800

In damon_young_pmd_entry(), it calls pmd_trans_huge() to check whether the
OS supports THP, if CONFIG_TRANSPARENT_HUGEPAGE is not included, the
pmd_trans_huge() will return 0.

Link: https://lkml.kernel.org/r/20220915142237.92529-2-xhao@linux.alibaba.com
Signed-off-by: Xin Hao <xhao@linux.alibaba.com>
Cc: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/vaddr.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/damon/vaddr.c~mm-damon-core-remove-duplicate-check-about-thp
+++ a/mm/damon/vaddr.c
@@ -434,7 +434,6 @@ static int damon_young_pmd_entry(pmd_t *
 	struct page *page;
 	struct damon_young_walk_private *priv = walk->private;
 
-#ifdef CONFIG_TRANSPARENT_HUGEPAGE
 	if (pmd_trans_huge(*pmd)) {
 		ptl = pmd_lock(walk->mm, pmd);
 		if (!pmd_present(*pmd)) {
@@ -462,7 +461,6 @@ huge_out:
 	}
 
 regular_page:
-#endif	/* CONFIG_TRANSPARENT_HUGEPAGE */
 
 	if (pmd_none(*pmd) || unlikely(pmd_bad(*pmd)))
 		return -EINVAL;
_
